### PR TITLE
Add ERC: Validation Module Extension for ERC-7579

### PR DESCRIPTION
**Pull request by kopy-kat**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: ERCS/erc-x.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
+---
```
Comment (abcoathup) [inline ERCS/erc-x.md:1]:
```suggestion
---
eip: 7780
```
Assigning next sequential EIP/ERC/RIP number.

Please also update the filename.

File: ERCS/erc-x.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+title: Validation Module Extension for ERC-7579
+description: Introduces three new module types for signature validation and permissioning
+author: zeroknots (@zeroknots), Konrad Kopp (@kopy-kat), Taek Lee (@leekt), Fil Makarov (@filmakarov)
+discussions-to: <URL>
```
Comment (abcoathup) [inline ERCS/erc-x.md:5]: Please create a discussions topic in Eth Magicians with a link to this PR: https://ethereum-magicians.org/c/ercs/57

File: ERCS/erc-7780.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
+---
+eip: 7780
+title: Validation Module Extension for ERC-7579
+description: Introduces three new module types for signature validation and permissioning
```
Comment (SamWilsn) [inline ERCS/erc-7780.md:4]: Just to make this more searchable, how about something like:

```suggestion
description: Introduces new smart account module types for signature validation and permissioning
```

File: ERCS/erc-7780.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+category: ERC
+created: 2024-10-01
+requires: 7579
+---
+
+## Abstract
+
+This proposal introduces three new module types on top of the existing modules described in [ERC-7579](./eip-7579). The modules are policy, signer and stateless validator. None of these modules are required to be implemented by accounts, but accounts can choose to implement them or other modules can choose to make use of them for additional composability.
+
+Policy modules can be used to check what a UserOperation or action is trying to achieve and determine if this is allowed. Signer modules can be used to validate signatures on provided hashes. Stateless validators are modules that are used to both validate signatures and compare them to a calldata-provided data blob which could, for example, include owners to check signatures against.
```
Comment (SamWilsn) [inline ERCS/erc-7780.md:18]: Code should be in backticks (not just here, but throughout the document.)

```suggestion
Policy modules can be used to check what a `UserOperation` or action is trying to achieve and determine if this is allowed. Signer modules can be used to validate signatures on provided hashes. Stateless validators are modules that are used to both validate signatures and compare them to a calldata-provided data blob which could, for example, include owners to check signatures against.
```

File: ERCS/erc-7780.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+- Policy (type id: 5)
+- Signer (type id: 6)
+- Stateless Validator (type id: 7)
+
+Note: A single module can be of multiple types.
+
+### Policy
+
+Policies MUST implement the `IModule` and the `IPolicy` interface and have module type id: `5`.
```
Comment (SamWilsn) [inline ERCS/erc-7780.md:42]:
```suggestion
Policies MUST implement [ERC-7579](./eip-7579.md)'s `IModule` and the `IPolicy` interface and have module type id: `5`.
```

File: ERCS/erc-7780.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+		/**
+		 * Checks a signature to determine if it should be executed
+		 * @param id The id of the policy
+		 * @param sender The sender of the transaction
+		 * @param hash The hash of the transaction
+		 * @param sig The signature of the transaction
+		 * @return The validation data to return to the EntryPoint
+		 */
+		function checkSignaturePolicy(
```
Comment (SamWilsn) [inline ERCS/erc-7780.md:70]: You might want to add some requirements around what this function does. Maybe something like:

> Policies MUST validate the given `hash` against `sig`. Policies MUST revert if the signature is invalid. Policies SHOULD verify that the sender is..."

I'm far from an expert, so obviously write requirements that make sense and don't just take my suggestion literally.

File: ERCS/erc-7780.md [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+		    bytes32 id,
+			  address sender,
+			  bytes32 hash,
+			  bytes calldata sig
+		)
+        external
+        view
+        virtual
+        returns (uint256);
```
Comment (SamWilsn) [inline ERCS/erc-7780.md:79]: Would be nice to have a bit more detail on what the return value is supposed to mean (even if it's just linking to the definition in another EIP.)

File: ERCS/erc-7780.md [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+     *
+     * MUST return true if the module is of the given type and false otherwise
+     */
+    function isModuleType(uint256 moduleTypeId) external view returns (bool);
+}
+```
+
+## Rationale
+
+TBD
```
Comment (SamWilsn) [inline ERCS/erc-7780.md:166]:
```suggestion
TBD <!-- TODO -->
```

The linter will make sure you replace any HTML-style comments before you go into Review.

File: ERCS/erc-7780.md [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+
+## Backwards Compatibility
+
+No backward compatibility issues found.
+
+## Security Considerations
+
+No security considerations found.
```
Comment (SamWilsn) [inline ERCS/erc-7780.md:174]:
```suggestion
TBD <!-- TODO -->
```

This is a signature validation proposal. There will absolutely be security considerations.

**Review by SamWilsn — APPROVED** [review]

**Pull request merged**